openslides_backend/action/actions/user/json_upload.py

```diff
@@ -0,0 +1,167 @@
+                    for field in ("username", "first_name", "last_name", "email")
+                }
+                for entry in data
+            ]
+        )
+        rows = [
+            self.generate_entry(entry, payload_index)
+            for payload_index, entry in enumerate(data)
+        ]
+
+        self.init_rows(rows)
+        self.store_rows_in_the_action_worker("account")
+        return {}
+
+    def generate_entry(
+        self, entry: Dict[str, Any], payload_index: int
+    ) -> Dict[str, Any]:
+        status, error = None, []
+        try:
+            UserCreate.schema_validator(entry)
+            if entry.get("username"):
+                if self.check_username_for_duplicate(entry["username"], payload_index):
+                    status = ImportStatus.DONE
+                    if searchdata := self.get_search_data(payload_index):
+                        entry["id"] = searchdata["id"]
+                    else:
+                        status = ImportStatus.ERROR
+                        error.append(f"Duplicate in csv list index: {payload_index}")
+                else:
+                    status = ImportStatus.NEW
+                entry["username"] = {"value": entry["username"], "info": "done"}
+            else:
+                if not (entry.get("first_name") or entry.get("last_name")):
+                    status = ImportStatus.ERROR
+                    error.append("Cannot generate username.")
+                elif self.check_name_and_email_for_duplicate(
+                    *_names_and_email(entry), payload_index
+                ):
+                    status = ImportStatus.DONE
+                    if searchdata := self.get_search_data(payload_index):
+                        entry["username"] = {
+                            "value": searchdata["username"],
+                            "info": "done",
+                        }
+                        entry["id"] = searchdata["id"]
+                    else:
+                        status = ImportStatus.ERROR
+                        error.append("Duplicate in csv list index: {payload_index}")
+                else:
+                    status = ImportStatus.NEW
+                    entry["username"] = {
+                        "value": self.generate_username(entry),
+                        "info": "generated",
+                    }
+            self.handle_default_password(entry, status)
+        except fastjsonschema.JsonSchemaException as exception:
+            status = ImportStatus.ERROR
+            error.append(exception.message)
+        return {"status": status, "error": error, "data": entry}
+
+    def generate_username(self, entry: Dict[str, Any]) -> str:
+        return self.generate_usernames(
+            [
+                re.sub(
+                    r"\W",
+                    "",
+                    entry.get("first_name", "") + entry.get("last_name", ""),
+                )
+            ]
+        )[0]
```

Comment:
Merge with method in `user.create` e.g. by moving it to the username mixin as `generate_username(first_name: str, last_name: str)`

Reply:
Done.